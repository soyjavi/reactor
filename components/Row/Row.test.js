import React from 'react';
import renderer from 'react-test-renderer';

import { Row } from '..';

describe('<Row>', () => {
  it('renders', () => {
    const tree = renderer.create(<Row />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {align}', () => {
    const tree = renderer.create(<Row align="start" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {justify}', () => {
    const tree = renderer.create(<Row justify="center" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('hook:styler', () => {
    const snapshot = renderer.create(<Row marginBottom="M" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
