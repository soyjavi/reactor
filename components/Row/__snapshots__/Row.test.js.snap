// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row> hook:styler 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "flex-start",
      },
      Object {
        "marginBottom": 16,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Row> renders 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
/>
`;

exports[`<Row> when {align} 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "alignItems": "flex-start",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
/>
`;

exports[`<Row> when {justify} 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;
